apps/src/publicKeyCryptography/ModuloClock.jsx

```diff
@@ -56,32 +60,35 @@ const ModuloClock = React.createClass({
     const maximumTimePerSegment = Math.min(1000, maximumDuration / 10);
 
     this.targetDividend = dividend;
-    this.startTime = Date.now();
     this.interval = setInterval(this.tick, 33);
     this.onStep = onStep || function () {};
     this.onComplete = onComplete || function () {};
     this.duration = Math.min(maximumDuration, dividend * maximumTimePerSegment);
-    this.setState({currentDividend: 0});
+    this.setState({startTime: Date.now(), currentDividend: 0});
   },
 
   tick() {
-    const elapsedTime = Date.now() - this.startTime;
+
+    const elapsedTime = Date.now() - this.state.startTime;
     if (elapsedTime < this.duration) {
       const currentDividend = Math.floor(easeOutCircular(elapsedTime, 0, this.targetDividend, this.duration));
```

Comment:
Animating the clock basically means interpolating the dividend from zero to a final value.  For very large dividends it's helpful to go faster at the beginning and slower as you approach the final value because you get a better look at how the final value is reached.  We want this deceleration to be smooth and to always complete in a certain amount of time.  In addition we probably don't want to render _every_ value between zero and the final dividend - for performance reasons we limit ourselves to rendering 30fps which means skipping a bunch of intermediate states when the animation is running fast.  So the question to ask for each rendered is: Given that I'm `n` seconds into an `m` second animation, and a I want a smooth deceleration, what intermediate dividend should I draw this time?

![image](https://cloud.githubusercontent.com/assets/1615761/22357209/86cff61e-e3eb-11e6-9a58-b021a000382a.png)

Enter `easeOutCircular` which is an [easing equation](http://gizma.com/easing/), a pure function representing a curve.  The curve is defined by a start and end value (here, `0` and `this.targetDividend`) and how long it should take to travel between those two values (here, `this.duration`).  When you give it a time value `elapsedTime` it returns the value between the start and end values that corresponds to that time in your animation - in our case, the dividend we should render on this frame.

![image](https://cloud.githubusercontent.com/assets/1615761/22356927/981ec014-e3e9-11e6-95b1-60601c554634.png)

That 'decelerating to _zero_ velocity' bit is what makes the animation feel extra slow at the end, and the strategy for cleaning that up is not generally to cap the duration of each step (which can interfere with the smoothness of the animation) but to cut the animation off a little early, before the velocity gets close to zero.  That's part of the change I made in the latest commit - I snap to the final value 200ms before the end of the curve now.